File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/TaskScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.Task;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class TaskScreenData {
	public static class Read extends ScreenData<Integer> {

		public Read(Integer params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<Task> {

		public Edit(Task task) {
			super(task);
		}
	}

	public static class Add extends ScreenData<Task> {

		public Add(Task task) {
			super(task);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/TaskScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.Task
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object TaskScreenData {

  class Read(param: Integer) extends ScreenData[Integer](param) {}

  class Edit(param: Task) extends ScreenData[Task](param) {}

  class Add(param: Task) extends ScreenData[Task](param) {}

  class GotoDashboard(query: Any) extends ScreenData[Any](query) {}

  class GotoKanbanView extends ScreenData {}

}
